controllers/nutanixmachine_controller.go

```diff
@@ -775,7 +775,23 @@ func getDiskList(rctx *nctx.MachineContext) ([]*prismclientv3.VMDisk, error) {
 }
 
 func getSystemDisk(rctx *nctx.MachineContext) (*prismclientv3.VMDisk, error) {
-	nodeOSImage, err := GetImage(rctx.Context, rctx.NutanixClient, rctx.NutanixMachine.Spec.Image)
+	var nodeOSImage *prismclientv3.ImageIntentResponse
+	var err error
+	if rctx.NutanixMachine.Spec.Image != nil {
+		nodeOSImage, err = GetImage(
+			rctx.Context,
+			rctx.NutanixClient,
+			rctx.NutanixMachine.Spec.Image,
+		)
+	} else if rctx.NutanixMachine.Spec.ImageLookup != nil {
+		nodeOSImage, err = GetImageWithFormat(
+			rctx.Context,
+			rctx.NutanixClient,
+			rctx.NutanixMachine.Spec.ImageLookup.Format,
+			&rctx.NutanixMachine.Spec.ImageLookup.BaseOS,
+			rctx.Machine.Spec.Version,
+		)
+	}
```

Comment:
What if both Image and ImageLookup are nil?

Reply:
this is an invalid state because of the CEL statement.